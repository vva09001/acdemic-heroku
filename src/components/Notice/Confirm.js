import React from "react";
import PropTypes from "prop-types";
import Dialog from "@material-ui/core/Dialog";
import { ButtonClick } from "../../theme/Button";

const ConfirmDialog = ({ handleClose, confirmAction, open, title }) => {
  return (
    <Dialog onClose={handleClose} aria-labelledby="confirm" open={open}>
      <div className="notice">
        <div className="noticeContent">{title}</div>
        <div className="noticeFooter">
          <ButtonClick
            classes="confirmBtn"
            onClick={confirmAction}
            title="Xác nhận"
          />
          <ButtonClick
            classes="cancelBtn"
            onClick={handleClose}
            title="Hủy bỏ"
          />
        </div>
      </div>
    </Dialog>
  );
};

ConfirmDialog.propTypes = {
  open: PropTypes.bool
};
ConfirmDialog.defaultProps = {
  open: false
};

export default ConfirmDialog;
