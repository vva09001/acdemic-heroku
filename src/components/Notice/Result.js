import React from "react";
import PropTypes from "prop-types";

import Dialog from "@material-ui/core/Dialog";
import { ButtonClick } from "../../theme/Button";

const ResultNotice = ({ handleClose, open, title }) => {
  return (
    <Dialog onClose={handleClose} aria-labelledby="success" open={open}>
      <div className="notice">
        <div className="noticeContent">{title}</div>
        <div className="noticeFooter">
          <ButtonClick
            classes="confirmBtn"
            onClick={handleClose}
            title="Xác nhận"
          />
        </div>
      </div>
    </Dialog>
  );
};

ResultNotice.propTypes = {
  open: PropTypes.bool
};

ResultNotice.defaultProps = {
  open: false
};
export default ResultNotice;
