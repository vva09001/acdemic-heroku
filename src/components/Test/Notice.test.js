import React from "react";
import { create } from "react-test-renderer";
import Confirm from "../Notice/Confirm";
import Result from "../Notice/Result";

describe("Notice Testing", () => {
  it("Test Result component it work", () => {
    const component = create(<Result />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender Search component`] = (
    <Result handleClose={[Function]} open={undefined} title={[Text]} />
  );
  it("Test Confirm component it work", () => {
    const component = create(<Confirm />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender Notice component`] = (
    <Confirm
      handleClose={[Function]}
      confirmAction={[Function]}
      open={undefined}
      title={[String]}
    />
  );
});
