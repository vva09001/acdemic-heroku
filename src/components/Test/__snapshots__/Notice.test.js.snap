// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Component Test Confirm component it work 1`] = `null`;

exports[`Confirm Component Test Confirm component it work 2`] = `null`;

exports[`Notice Testing Test Confirm component it work 1`] = `null`;

exports[`Notice Testing Test Result component it work 1`] = `null`;
